shipment_advice_reception_planner/models/stock_move.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2024 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    origin_purchase = fields.Char(
+        compute="_compute_origin_purchase", search="_search_origin_purchase"
+    )
+    quantity_to_split_wizard = fields.Float(
+        string="Split Demand", store=False, readonly=False
+    )
+
+    @api.depends("purchase_line_id")
+    def _compute_origin_purchase(self):
+        for move in self:
+            if not move.purchase_line_id:
+                move.origin_purchase = False
+                continue
+            purchase = move.purchase_line_id.order_id
+            move.origin_purchase = " ".join(
+                [
+                    origin
+                    for origin in [
+                        purchase.name,
+                        purchase.origin,
+                    ]
+                    if origin
+                ]
+            )
+
+    def _search_origin_purchase(self, operator, value):
+        purchases = self.env["purchase.order"].search(
+            ["|", ("name", operator, value), ("origin", operator, value)]
+        )
+        return [("purchase_line_id", "in", purchases.order_line.ids)]
+
+    @api.onchange("quantity_to_split_wizard")
+    def onchange_quantity_to_split(self):
```

Comment:
Onchanges should be private methods with '_'. Moreover, why not using constraints ?